<?xml version="1.0"?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from mmrobot.urdf.xacro             | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="albert">
  <link name="world"/>
  <joint name="boxer_joint_world" type="fixed">
    <parent link="world"/>
    <child link="odom"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <link name="odom"/>
  <link name="world_base_link_x"/>
  <link name="world_base_link_y"/>
  <link name="base_link_ori"/>
  <joint name="world_base_x" type="prismatic">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="odom"/>
    <child link="world_base_link_x"/>
    <axis xyz="1 0 0"/>
    <limit effort="100" lower="-15" upper="15" velocity="5"/>
  </joint>
  <joint name="world_base_y" type="prismatic">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="world_base_link_x"/>
    <child link="world_base_link_y"/>
    <axis xyz="0 1 0"/>
    <limit effort="100" lower="-15" upper="15" velocity="5"/>
  </joint>
  <joint name="world_base_theta" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="world_base_link_y"/>
    <child link="base_link_ori"/>
    <axis xyz="0 0 1"/>
    <limit effort="100" lower="-15" upper="15" velocity="5"/>
  </joint>
  <joint name="base_add_orientation" type="fixed">
    <origin rpy="0 0 1.57079" xyz="0.0 0 0"/>
    <parent link="base_link_ori"/>
    <child link="base_link"/>
  </joint>
  <link name="chassis_link"/>
  <link name="internal_link"/>
  <joint name="internal_joint" type="fixed">
    <origin rpy="1.5708 0 1.5708" xyz="0 0 0"/>
    <parent link="chassis_link"/>
    <child link="internal_link"/>
    <axis xyz="0 0 0"/>
  </joint>
  <link name="lift_link"/>
  <joint name="lift_joint" type="fixed">
    <origin rpy="1.5708 0 1.5708" xyz="0 0 0.177"/>
    <parent link="chassis_link"/>
    <child link="lift_link"/>
    <axis xyz="0 1 0"/>
    <limit effort="10000.0" lower="0" upper="0.078" velocity="0.02"/>
    <dynamics damping="100.0" friction="100"/>
  </joint>
  <link name="top_mount_bottom">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="10"/>
      <inertia ixx="0.50" ixy="0" ixz="0" iyy="0.50" iyz="0" izz="1.00"/>
    </inertial>
  </link>
  <joint name="top_mount_bottom_joint" type="fixed">
    <origin rpy="0 -1.5707 -1.5707" xyz="0 0.1125 0"/>
    <parent link="lift_link"/>
    <child link="top_mount_bottom"/>
  </joint>
  <link name="extrusion1"/>
  <joint name="extrusion1_joint" type="fixed">
    <origin rpy="0 0 0" xyz="-0.2785 0.1835 0"/>
    <parent link="top_mount_bottom"/>
    <child link="extrusion1"/>
  </joint>
  <link name="top_mount"/>
  <joint name="top_mount_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0.2785 -0.1835 0.31"/>
    <parent link="extrusion1"/>
    <child link="top_mount"/>
  </joint>
  <link name="base_link"/>
  <link name="base_tip_link"/>
  <joint name="base_tip_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0.4 0 0"/>
    <parent link="chassis_link"/>
    <child link="base_tip_link"/>
  </joint>
  <joint name="base_chassis_joint" type="fixed">
    <origin rpy="0 0 -1.5708" xyz="0.0 0 0"/>
    <parent link="base_link"/>
    <child link="chassis_link"/>
  </joint>
  <joint name="panda_joint_top_mount" type="fixed">
    <parent link="top_mount"/>
    <child link="panda_link0"/>
    <origin rpy="0 0 0" xyz="0.387 0 0"/>
  </joint>
  <link name="panda_link0"/>
  <link name="panda_link1"/>
  <link name="panda_link2"/>
  <link name="panda_link3"/>
  <link name="panda_link4"/>
  <link name="panda_link5"/>
  <link name="panda_link6"/>
  <link name="panda_link7"/>
  <link name="panda_link8"/>
  <link name="panda_link9"/>
  <joint name="panda_joint1" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <origin rpy="0 0 0" xyz="0 0 0.333"/>
    <parent link="panda_link0"/>
    <child link="panda_link1"/>
    <axis xyz="0 0 1"/>
    <limit effort="87" lower="-2.8973" upper="2.8973" velocity="2.1750"/>
  </joint>
  <joint name="panda_joint2" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-1.7628" soft_upper_limit="1.7628"/>
    <origin rpy="-1.57079632679 0 0" xyz="0 0 0"/>
    <parent link="panda_link1"/>
    <child link="panda_link2"/>
    <axis xyz="0 0 1"/>
    <limit effort="87" lower="-1.7628" upper="1.7628" velocity="2.1750"/>
  </joint>
  <joint name="panda_joint3" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <origin rpy="1.57079632679 0 0" xyz="0 -0.316 0"/>
    <parent link="panda_link2"/>
    <child link="panda_link3"/>
    <axis xyz="0 0 1"/>
    <limit effort="87" lower="-2.8973" upper="2.8973" velocity="2.1750"/>
  </joint>
  <joint name="panda_joint4" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-3.0718" soft_upper_limit="0.0698"/>
    <origin rpy="1.57079632679 0 0" xyz="0.0825 0 0"/>
    <parent link="panda_link3"/>
    <child link="panda_link4"/>
    <axis xyz="0 0 1"/>
    <limit effort="87" lower="-3.0718" upper="0.0698" velocity="2.1750"/>
  </joint>
  <link name="panda_link5_offset">
    </link>
  <joint name="panda_joint5_offset" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 -0.20"/>
    <parent link="panda_link5"/>
    <child link="panda_link5_offset"/>
  </joint>
  <joint name="panda_joint5" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <origin rpy="-1.57079632679 0 0" xyz="-0.0825 0.384 0"/>
    <parent link="panda_link4"/>
    <child link="panda_link5"/>
    <axis xyz="0 0 1"/>
    <limit effort="12" lower="-2.8973" upper="2.8973" velocity="2.6100"/>
  </joint>
  <joint name="panda_joint6" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-0.0175" soft_upper_limit="3.7525"/>
    <origin rpy="1.57079632679 0 0" xyz="0 0 0"/>
    <parent link="panda_link5"/>
    <child link="panda_link6"/>
    <axis xyz="0 0 1"/>
    <limit effort="12" lower="-0.0175" upper="3.7525" velocity="2.6100"/>
  </joint>
  <joint name="panda_joint7" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <origin rpy="1.57079632679 0 0" xyz="0.088 0 0"/>
    <parent link="panda_link6"/>
    <child link="panda_link7"/>
    <axis xyz="0 0 1"/>
    <limit effort="12" lower="-2.8973" upper="2.8973" velocity="2.6100"/>
  </joint>
  <joint name="panda_joint8" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.107"/>
    <parent link="panda_link7"/>
    <child link="panda_link8"/>
    <axis xyz="0 0 0"/>
  </joint>
  <joint name="panda_joint9" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.020"/>
    <parent link="panda_link8"/>
    <child link="panda_link9"/>
    <axis xyz="0 0 0"/>
  </joint>
  <joint name="panda_hand_joint" type="fixed">
    <parent link="panda_link8"/>
    <child link="panda_hand"/>
    <origin rpy="0 0 -0.785398163397" xyz="0 0 0"/>
  </joint>
  <link name="panda_hand"/>
  <link name="panda_leftfinger"/>
  <link name="panda_rightfinger"/>
  <joint name="panda_finger_joint1" type="fixed">
    <parent link="panda_hand"/>
    <child link="panda_leftfinger"/>
    <origin rpy="0 0 0" xyz="0 0 0.0584"/>
    <axis xyz="0 1 0"/>
    <limit effort="20" lower="0.0" upper="0.04" velocity="0.2"/>
  </joint>
  <joint name="panda_finger_joint2" type="fixed">
    <parent link="panda_hand"/>
    <child link="panda_rightfinger"/>
    <origin rpy="0 0 0" xyz="0 0 0.0584"/>
    <axis xyz="0 -1 0"/>
    <limit effort="20" lower="0.0" upper="0.04" velocity="0.2"/>
  </joint>
  <joint name="panda_vacuum_joint" type="fixed">
    <parent link="panda_hand"/>
    <child link="panda_vacuum"/>
    <origin rpy="0 0 0" xyz="0 0 0.33"/>
  </joint>
  <joint name="panda_vacuum_joint_2" type="fixed">
    <parent link="panda_hand"/>
    <child link="panda_vacuum_2"/>
    <origin rpy="0 0 0" xyz="0 0 0.25"/>
  </joint>
  <link name="panda_vacuum">
    </link>
  <link name="panda_vacuum_2">
    </link>
  <link name="panda_rightfinger">
    <inertial>
      <mass value="0.015"/>
      <inertia ixx="2.375e-06" ixy="0" ixz="0" iyy="2.375e-06" iyz="0" izz="7.5e-07"/>
    </inertial>
  </link>
  <link name="panda_leftfinger">
    <inertial>
      <mass value="0.015"/>
      <inertia ixx="2.375e-06" ixy="0" ixz="0" iyy="2.375e-06" iyz="0" izz="7.5e-07"/>
    </inertial>
  </link>
  <joint name="panda_finger_joint1" type="prismatic">
    <parent link="panda_hand"/>
    <child link="panda_leftfinger"/>
    <origin rpy="0 0 0" xyz="0 0 0.0584"/>
    <axis xyz="0 1 0"/>
    <limit effort="100" lower="0.0" upper="0.04" velocity="0.2"/>
    <dynamics damping="0.03" friction="0.0"/>
  </joint>
  <joint name="panda_finger_joint2" type="prismatic">
    <parent link="panda_hand"/>
    <child link="panda_rightfinger"/>
    <origin rpy="0 0 0" xyz="0 0 0.0584"/>
    <axis xyz="0 -1 0"/>
    <limit effort="100" lower="0.0" upper="0.04" velocity="0.2"/>
    <dynamics damping="0.03" friction="0.0"/>
    <mimic joint="panda_finger_joint1"/>
  </joint>
  <material name="aluminum">
    <color rgba="0.5 0.5 0.5 1"/>
  </material>
  <material name="plastic">
    <color rgba="0.1 0.1 0.1 1"/>
  </material>
  <!-- <xacro:include filename="$(find realsense2_description)/urdf/_usb_plug.urdf.xacro" /> -->
  <!-- new root link because we calibrated w.r.t. this -->
  <joint name="camera_calibration" type="fixed">
    <origin rpy="-0.08220003 -0.01235345  2.33171179" xyz="0.1052022 0.04861002 0.02188261"/>
    <parent link="panda_link8"/>
    <child link="camera_color_optical_frame"/>
  </joint>
  <link name="camera_color_optical_frame"/>
  <joint name="camera_color_optical_joint" type="fixed">
    <origin rpy="1.57079632679 -1.57079632679 0" xyz="0.015 0 0"/>
    <parent link="camera_color_optical_frame"/>
    <child link="camera_link"/>
  </joint>
  <link name="camera_link">
    <visual>
      <!-- the mesh origin is at front plate in between the two infrared camera axes -->
      <origin rpy="1.57079632679 0 1.57079632679" xyz="0.0043 -0.0175 0"/>
      <geometry>
        <mesh filename="package://realsense2_description/meshes/d435.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 -0.0175 0"/>
      <geometry>
        <box size="0.02505 0.09 0.025"/>
      </geometry>
    </collision>
    <!-- <inertial> -->
    <!-- The following are not reliable values, and should not be used for modeling -->
    <!-- <mass value="0.076" /> -->
    <!-- <origin xyz="0 0 0" /> -->
    <!-- <inertia ixx="0.003881243" ixy="0.0" ixz="0.0" iyy="0.000498940" iyz="0.0" izz="0.003879257" /> -->
    <!-- </inertial> -->
  </link>
  <!-- camera depth joints and links -->
  <joint name="camera_depth_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="camera_link"/>
    <child link="camera_depth_frame"/>
  </joint>
  <link name="camera_depth_frame"/>
  <joint name="camera_depth_optical_joint" type="fixed">
    <origin rpy="-1.57079632679 0 -1.57079632679" xyz="0 0 0"/>
    <parent link="camera_depth_frame"/>
    <child link="camera_depth_optical_frame"/>
  </joint>
  <link name="camera_depth_optical_frame"/>
  <!-- camera left IR joints and links -->
  <joint name="camera_infra1_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0.0 0"/>
    <parent link="camera_link"/>
    <child link="camera_infra1_frame"/>
  </joint>
  <link name="camera_infra1_frame"/>
  <joint name="camera_infra1_optical_joint" type="fixed">
    <origin rpy="-1.57079632679 0 -1.57079632679" xyz="0 0 0"/>
    <parent link="camera_infra1_frame"/>
    <child link="camera_infra1_optical_frame"/>
  </joint>
  <link name="camera_infra1_optical_frame"/>
  <!-- camera right IR joints and links -->
  <joint name="camera_infra2_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 -0.05 0"/>
    <parent link="camera_link"/>
    <child link="camera_infra2_frame"/>
  </joint>
  <link name="camera_infra2_frame"/>
  <joint name="camera_infra2_optical_joint" type="fixed">
    <origin rpy="-1.57079632679 0 -1.57079632679" xyz="0 0 0"/>
    <parent link="camera_infra2_frame"/>
    <child link="camera_infra2_optical_frame"/>
  </joint>
  <link name="camera_infra2_optical_frame"/>
</robot>

